### PR TITLE
remove if statement in ci.yml

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Remove if-statement since we donot have pipeline runs for various engsys related PRs that rely on the template tests as our primary CI gate.

### CODE REVIEW

File: sdk/template/ci.yml [hunk 1]
```diff
@@ -9,12 +9,7 @@ trigger:
   paths:
     include:
     - sdk/template/
-#if (false)
```
Comment (sima-zhu) [inline sdk/template/ci.yml:12]: Don't remove the if block. It is useful comment section.

**Review by sima-zhu — DISMISSED** [review]
Could you please add the trigger of "eng/common" under pr triggers as well. It is at #L23

File: sdk/template/ci.yml (hunk 1 above)
Comment (weshaggard) [reply sdk/template/ci.yml:12]: In fact we need to add back the if block that was removed as part of your other PR. @chunyu3 is there some reason you are trying to remove these?

Comment (weshaggard) [reply sdk/template/ci.yml:12]: See my comment in your other PR https://github.com/Azure/azure-sdk-for-net/pull/28016#discussion_r852480070.

**Review by benbp — APPROVED** [review]

File: sdk/template/ci.yml (hunk 1 above)
Comment (chunyu3) [reply sdk/template/ci.yml]: I updated according to the comment and add eng/common in pr-trigger the same as trigger does

**Pull request merged**